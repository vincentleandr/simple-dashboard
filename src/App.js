import React, { Component } from 'react';
import {
    BrowserRouter,
    Route,
    Switch
} from 'react-router-dom';
import axios from 'axios';

import './styles/styles.min.css';

import Header from './components/header';
import Sidenav from './components/sidenav';

import Home from './containers/home';
import NestedTab from './containers/nestedtab';
import List from './containers/list';

class App extends Component {
	constructor(props) {
        super(props);

        this.state = {
			name: 'Guest',
			lists: []
		}
		
		this.changeName = this.changeName.bind(this);
	}

	// Get API, push to state
	componentDidMount() {
		axios.get('https://jsonplaceholder.typicode.com/posts')
		.then(response => {
			this.setState({ 
				lists: response.data 
			});
		})
	}
	
	// Change user's name based on input value
	changeName = (inputName) => {
		this.setState({
			name: inputName
		});
	}

	render() {
		return (
			<div className="App">
				<Header />
				
				<BrowserRouter>
					<div className="body">
						<Sidenav />
						<div className="main-content">
							<Switch>
								<Route exact path="/" render={() => <Home name={this.state.name} changeName={this.changeName} />} />
								<Route path="/nestedtab" render={() => <NestedTab name={this.state.name} />} />
								<Route path="/list" render={() => <List lists={this.state.lists} />} />
							</Switch>
						</div>
					</div>
				</BrowserRouter>

			</div>
		);
	}
}

export default App;
