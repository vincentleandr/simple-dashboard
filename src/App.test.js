import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'

import App from './App';
import Header from './components/header';
import Sidenav from './components/sidenav';
import Home from './containers/home';
import NestedTab from './containers/nestedtab';
import List from './containers/list';

configure({ adapter: new Adapter() });


describe('App', () => {
  const wrapper = shallow(<App />);
  const header = <Header />;
  const sidenav = <Sidenav />;
  const home = <Home />;
  const nestedtab = <NestedTab />
  const list = <List />

  it('matches the snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('App renders without crashing', () => {
    wrapper;
  });

  it('App renders the header', () => {
    expect(wrapper.contains(header)).toEqual(true);
  });

  it('App renders the sidenav', () => {
    expect(wrapper.contains(sidenav)).toEqual(true);
  });

  it('App renders the home', () => {
    expect(wrapper.find(home));
  });

  it('always renders the nestedtab', () => {
    expect(wrapper.find(nestedtab));
  });

  it('always renders the list', () => {
    expect(wrapper.find(list));
  });

  it('App renders the maincontent and everything else inside it', () => {
    mount(<div className="main-content"></div>);
  });

  it('App have states', () => {
    expect(wrapper.state('name')).toEqual('Guest');
    expect(wrapper.state('lists')).toEqual([]);
  });

  it('Home receive correct name props', () => {
    const homeProps = shallow(<Home name='Guest' />);
    expect(homeProps.find('h1').text()).toEqual('Welcome, Guest !');
  });

})
