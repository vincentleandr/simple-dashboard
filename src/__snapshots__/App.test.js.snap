// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<div
  className="App"
>
  <div
    className="header"
  >
    <span>
      Simple Dashboard App
    </span>
  </div>
  <div
    className="body"
  >
    <nav
      className="sidenav"
    >
      <ul>
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            <span>
              Home
            </span>
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/nestedtab"
            onClick={[Function]}
          >
            <span>
              Nested Tab
            </span>
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/list"
            onClick={[Function]}
          >
            <span>
              List
            </span>
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="main-content"
    >
      <div
        className="home"
      >
        <h1>
          Welcome, 
          Guest
           !
        </h1>
        <input
          onChange={[Function]}
          placeholder="Type your name here..."
          type="text"
        />
      </div>
    </div>
  </div>
</div>
`;
